### PR TITLE
Format tables with numbers

### PR DESCRIPTION
**Pull request by 8ctopus**
- Add number formatting option
- Add left padding option
#3

### CODE REVIEW

**Comment by 8ctopus** [comment]
Here's an example of what the change brings

before
```txt
+--------+
| sum    |
+--------+
| 10.999 |
| 222    |
| 7      |
| 0      |
+--------+
```

after

```txt
+--------+
|    sum |
+--------+
|  11.00 |
| 222.00 |
|   7.00 |
|   0.00 |
+--------+
```

**Comment by 8ctopus** [comment]
Hello Denis, are you still working on the project? If not, no problem, I will publish my fork on composer.

**Comment by deniskoronets** [comment]
@8ctopus sorry for late response. I'll have a look into this PR today. Just using gitlab on my job so rarely visit github.

File: src/ArrayToTextTable.php [hunk 1]
```diff
@@ -13,42 +13,57 @@ class ArrayToTextTable
...
-    
+
     /**
      * @var bool
      */
     private $renderHeader = true;
-    
-    public function __construct(array $data)
+
+    public function __construct(array $data, ?array $format = null)
```
Comment (deniskoronets) [inline src/ArrayToTextTable.php:57]: would be MUCH better to add ?array $options = [] so we could reuse it later and don't break backward compatibility. @8ctopus could you please fix it? I'll merge it then and push a new tag to github

File: src/ArrayToTextTable.php [hunk 2]
```diff
@@ -13,42 +13,57 @@ class ArrayToTextTable
      * @var array
      */
     private $data;
-    
+
+    /**
+     * @var bool
+     */
+    private $paddingRight = true;
```
Comment (deniskoronets) [inline src/ArrayToTextTable.php:20]: add $options config instead

**Review by deniskoronets — CHANGES_REQUESTED** [review]

File: src/ArrayToTextTable.php (hunk 2 above)
Comment (8ctopus) [reply src/ArrayToTextTable.php]: I've moved the padding inside the format array.

File: src/ArrayToTextTable.php (hunk 1 above)
Comment (8ctopus) [reply src/ArrayToTextTable.php:52]: I've done it slightly differently because the $format array should not be empty.

**Comment by 8ctopus** [comment]
Hello Denis, any chance you could review the PR?

**Comment by 8ctopus** [comment]
If you don't have time, no problem, I will just detach the fork and publish it on packagist.

**Comment by deniskoronets** [comment]
hello! I'd do some changes to yor PR once I get some time. Hope to do it this week. Just have lots of work :(

**Comment by 8ctopus** [comment]
Ok Denis, thank you.

**Comment by deniskoronets** [comment]
@8ctopus I'm working on an update to this library including your requested features. going to organize it in bit different way. you can track it here: https://github.com/deniskoronets/php-array-table/pull/5/files
going to complete it next week

**Comment by 8ctopus** [comment]
As long as it has the same features, it's fine with me.

**Comment by deniskoronets** [comment]
No longer needed as its implemented in different way:
https://packagist.org/packages/dekor/php-array-table
please check the doc

**Pull request closed without merging**